### PR TITLE
Note exact condition for determining individual pk

### LINKED ISSUE #80: Add more precise condition under which tweaking the individual pubkey is dangerous
"If the public key is not determined before sending out the pubnonce, an adversary can forge a signature by influencing the individual tweak of a victim signer after seeing the pubnonce "

I want to add a footnote that specifies a more precise condition for future reference (if someone wants to tweak the pubkey and it's not determined yet... which is ok in some cases).

### PR DESCRIPTION
**Pull request by real-or-random**
Fixes #80.

Well, after having it written down now, I'm more than confident that what we did in the (main body of the) BIP is better in practice.

### CODE REVIEW

File: bip-musig2.mediawiki [hunk 1]
```diff
@@ -535,8 +535,20 @@ Implementers must avoid modifying the ''NonceGen'' algorithm without being fully
...
+More formally, consider modified algorithms ''NonceGen' '' and ''Sign' '', where ''NonceGen' '' does not take the individual public key of the signer as input and does not store it in pubnonce, and Sign' does not check read the individual public key from pubnonce and does not check it against the secret key taken as input.
+Then it suffices that for each invocation of ''NonceGen' '' with output ''(secnonce, pubnonce)'',
+a function ''fsk'' is determined before sending out ''pubnonce'',
+where ''fsk'' maps a pair consisting of a list of plain public keys and a message to a secret key,
+such that the secret key ''sk'' and the session context ''session_ctx''
+passed to the corresponding invocation of ''Sign'(secnonce, sk, session_ctx)'',
+adhere to the condition ''fsk(pk<sub>1..u</sub>, m) = sk'',
+where ''(_, _, pk<sub>1..u</sub>, _, _, _, m) = session_ctx''.<br />
```
Comment (jonasnick) [inline bip-musig2.mediawiki]:
```suggestion
such that the secret key ''sk'' and the session context ''session_ctx = (_, _, pk<sub>1..u</sub>, _, _, _, m)''
passed to the corresponding invocation of ''Sign'(secnonce, sk, session_ctx)'',
adhere to the condition ''fsk(pk<sub>1..u</sub>, m) = sk''.<br />
```

File: bip-musig2.mediawiki [hunk 2]
```diff
@@ -535,8 +535,20 @@ Implementers must avoid modifying the ''NonceGen'' algorithm without being fully
...
+Thus, implementations must guarantee that the public key of the signer is determined before sending out the ''pubnonce'' if they cannot rule out that the signer's individual key is tweaked (e.g., derived via BIP32) with a tweak known to the adversary (e.g., as in BIP32 unhardened derivation).<ref>The nonce generation algorithm as given in the [https://eprint.iacr.org/2020/1261 MuSig2 paper] does not take ''pk'' as an input (and does not store it in the ''secnonce''). However, the security model considered in the paper does not cover security for signers who tweak their individual keys.</ref><ref>In fact, it is not a problem that the attacker can influence the individual tweak of a victim signer after seeing the ''pubnonce'' if the signer uses nothing but the list of the public keys of all signers and the message to sign for determining what tweak to use.<br />
```
Comment (jonasnick) [inline bip-musig2.mediawiki]: nit: tbh I only understood the purpose of the sentence (and of the paragraph) after having read the full paragraph. Maybe we could more explicitly say that there are alternatives to our choice (which is the reason for having this note in the BIP I guess). Maybe something like "In fact, the public key does is not required to be uniquely determined before sending out the `pubnonce`. It is not a problem [...]"

Comment (real-or-random) [reply bip-musig2.mediawiki]: Yes, you're right. I changed it, it should be much better now.

**Comment by real-or-random** [comment]
@jonasnick I tried to change it in the spirit that we've discussed. Please have a look.

**Review by jonasnick — COMMENTED** [review]
Nice, looks good!

**Pull request merged**